script.lua

```diff
@@ -6349,37 +6337,87 @@ function setupColor(table, color)
         colorTint = Color.fromHex(Tints[color].Presence)
     end
     table.setColorTint(colorTint)
-    table.UI.setXml("")
     table.clearButtons()
-    setupSwapButtons(table)
+    setupSwapButtons(table, { seated = true })
 
     updateColorPickButtons()
-    updateSwapButtons()
 end
-function setupSwapButtons(obj)
-    local scale = flipVector(Vector(obj.getScale()))
-    scale = scale * 2
-    -- Swap Place (button index 0)
-    obj.createButton({
-        label="", click_function="onClickedSwapPlace", function_owner=Global,
-        position={-3.25,0.4,7.5}, rotation={0,0,0}, height=0, width=0, scale=scale,
-        font_color={0,0,0}, font_size=250,
-        tooltip="Moves your current player color to be located here. The color currently seated here will be moved to your current location. Spirit panels and other cards will be relocated if applicable.",
+-- @param params.seated Boolean indicating if a player is about to be seated here.
+function setupSwapButtons(obj, params)
```

Comment:
if params table is only every going to have a boolean, why have this parameter be a table?

Reply:
Because it is slightly clearer to understand what `true` means in 
```
setupSwapButtons(table, { seated = true })
```
than in
```
setupSwapButtons(table, true)
```

Very often, understanding what boolean parameters mean is hard to do in the context of a function call. Named parameters make this easier to do, and this is one way to get that kind of behaviour in lua.